pandas/tseries/holiday.py

```diff
@@ -156,8 +156,8 @@ class from pandas.tseries.offsets
         self.month = month
         self.day = day
         self.offset = offset
-        self.start_date = start_date
-        self.end_date = end_date
+        self.start_date = Timestamp(start_date) if start_date is not None else start_date
+        self.end_date = Timestamp(end_date) if end_date is not None else end_date
```

Comment:
do we need a ``closed='left|right|both`` parameter to make this non-ambiguous? (default prob both)?